File: tourguide/src/main/java/tourguide/tourguide/Pointer.java
package tourguide.tourguide;

import android.graphics.Color;
import android.view.Gravity;

/**
 * Created by tanjunrong on 6/20/15.
 */
public class Pointer {
    public int mGravity = Gravity.CENTER;
    public int mColor = Color.WHITE;

    public Pointer() {
        this(Gravity.CENTER, Color.parseColor("#FFFFFF"));
    }

    public Pointer(int gravity, int color) {
        this.mGravity = gravity;
        this.mColor = color;
    }

    /**
     * Set color
     * @param color
     * @return return Pointer instance for chaining purpose
     */
    public Pointer setColor(int color){
        mColor = color;
        return this;
    }

    /**
     * Set gravity
     * @param gravity
     * @return return Pointer instance for chaining purpose
     */
    public Pointer setGravity(int gravity){
        mGravity = gravity;
        return this;
    }
}

File: tourguide/src/main/java/tourguide/tourguide/Pointer.kt
package tourguide.tourguide

import android.graphics.Color
import android.view.Gravity

class Pointer @JvmOverloads constructor(
        var gravity: Int = Gravity.CENTER,
        var color: Int = Color.parseColor("#FFFFFF")
) {

    var technique: Technique = Technique.CLICK

    fun gravity(block: () -> Int) {
        gravity = block()
    }

    fun color(block: () -> Int) {
        color = block()
    }

    fun technique(block: () -> Technique) {
        technique = block()
    }

    /**
     * This describes the animation techniques
     */
    enum class Technique {
        CLICK, HORIZONTAL_LEFT, HORIZONTAL_RIGHT, VERTICAL_UPWARD, VERTICAL_DOWNWARD
    }
}